frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -114,57 +115,73 @@ type cvdInstance struct {
 	InstanceDir  string   `json:"instance_dir"`
 }
 
-func cvdFleet(ctx cvd.CVDExecContext, cvdBin string) ([]*cvdInstance, error) {
+func getCVDFleet(ctx cvd.CVDExecContext, cvdBin string) (cvdFleet, error) {
 	stdout := &bytes.Buffer{}
 	cvdCmd := cvd.NewCommand(ctx, cvdBin, []string{"fleet"}, cvd.CommandOpts{Stdout: stdout})
-	err := cvdCmd.Run()
-	if err != nil {
-		return nil, err
+	if err := cvdCmd.Run(); err != nil {
+		return cvdFleet{}, err
 	}
-	output := &cvdFleetOutput{}
-	if err := json.Unmarshal(stdout.Bytes(), output); err != nil {
+	result := cvdFleet{}
+	if err := json.Unmarshal(stdout.Bytes(), &result); err != nil {
 		log.Printf("Failed parsing `cvd fleet` ouput. Output: \n\n%s\n", cvd.OutputLogMessage(stdout.String()))
-		return nil, fmt.Errorf("failed parsing `cvd fleet` output: %w", err)
+		return cvdFleet{}, fmt.Errorf("failed parsing `cvd fleet` output: %w", err)
 	}
-	if len(output.Groups) == 0 {
-		return []*cvdInstance{}, nil
-	}
-	// Host orchestrator only works with one instances group.
-	return output.Groups[0].Instances, nil
+	return result, nil
 }
 
-func fleetToCVDs(val []*cvdInstance) []*apiv1.CVD {
-	result := make([]*apiv1.CVD, len(val))
-	for i, item := range val {
-		result[i] = &apiv1.CVD{
-			Name: item.InstanceName,
-			// TODO(b/259725479): Update when `cvd fleet` prints out build information.
-			BuildSource: &apiv1.BuildSource{},
-			Status:      item.Status,
-			Displays:    item.Displays,
+func fleetToCVDs(fleet cvdFleet) []*apiv1.CVD {
+	result := []*apiv1.CVD{}
+	for _, group := range fleet.Groups {
+		for _, ins := range group.Instances {
+			cvd := &apiv1.CVD{
+				Name: encodeCVDName(group.Name, ins.InstanceName),
+				// TODO(b/259725479): Update when `cvd fleet` prints out build information.
+				BuildSource: &apiv1.BuildSource{},
+				Status:      ins.Status,
+				Displays:    ins.Displays,
+			}
+			result = append(result, cvd)
 		}
 	}
 	return result
 }
 
-func CVDLogsDir(ctx cvd.CVDExecContext, cvdBin, name string) (string, error) {
-	instances, err := cvdFleet(ctx, cvdBin)
+// Use same name format used in `GET /devices` API.
+func encodeCVDName(group, name string) string {
+	return group + "-" + name
+}
+
+func decodeCVDName(name string) (string, string, error) {
+	parts := strings.SplitN(name, "-", 2)
+	if len(parts) != 2 {
+		return "", "", errors.New("invalid <GROUP>-<NAME> format")
+	}
+	return parts[0], parts[1], nil
+}
```

Comment:
Don't do this, the latest version of `cvd` already prints instance name, group name and webrtc device id separately.

Reply:
I see, I'll abandon this PR in favor of using `webrtc device id`: https://github.com/google/android-cuttlefish/pull/355